src/models/document/base-document-content.ts

```diff
@@ -585,7 +585,7 @@ export const BaseDocumentContentModel = RowList.named("BaseDocumentContent")
       });
       // fix any "collapsed" sections
       self.allRowLists.forEach(rowList => {
-        for (let i = 1; i < rowList.rowCount; ++i) {
+        for (let i = 1; i <= rowList.rowCount; ++i) {
```

Comment:
I did a search using this regex (`<[^=](.*)rowCount`) on my current `master` based branch and I saw other uses of it in the codebase, all in this file.  I then checked them against the current raw contents of this branch here:

https://raw.githubusercontent.com/concord-consortium/collaborative-learning/ea8691d388e25834944254740ae34d007469c8ad/src/models/document/base-document-content.ts

to see if they still exist and they do.

The one in `removeNeighboringPlaceholderRows` seems fine as we do `rowCount - 1`.

The one in `addPlaceholderRowIfAppropriate` needs to change to `<=` though, it looks like this:

`const afterRow = (rowIndex < rowList.rowCount) && rowList.getRowByIndex(rowIndex);`

Reply:
Hm, I disagree.  I think `rowList.getRowByIndex(rowIndex)` will fail if `rowIndex===rowList.rowCount`.

We can call `addPlaceholderRowIfAppropriate` with `rowIndex===rowList.rowCount`, to ask it to check whether to add a placeholder after the last existing row.  But in that situation `afterRow` should be undefined.

I added a comment that may help clarify.